tools/diff_collator.py

```diff
@@ -0,0 +1,165 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Output a list of differences between current git branch and master
+
+Precondition: this script is run instide an existing git repository
+
+This script will performs a retrieves and output a list of changes
+based on the output of git diff. For each changes, the file, line numbers
+and top-level funtcion name is provided.
```

Comment:
typo: funtcion
Plus this is a bit hard to understand, please re-phrase these a bit.